Api/Helpers/SoapHelper.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.ServiceModel;
+using System.Threading.Tasks;
+using Api.Services;
+using HelloSoap;
+using Microsoft.Extensions.Logging;
+
+namespace Api.Helpers
+{
+    public static class SoapHelper {
+        /// <summary>
+        ///     Wraps SOAP logic related to opening/closing channels and exception handling.
+        /// </summary>
+        /// <typeparam name="TResult">Return type of provided delegate.</typeparam>
+        /// <param name="client">Pre-configured SOAP Client.</param>
+        /// <param name="action">Delegate containing logic to be executed whilst channel is open.</param>
+        /// <param name="logger">Logger (optional).</param>
+        /// <returns>Return value of the provided delegate.</returns>
+        public static async Task<TResult> IssueSoapCallAsync<TResult>(Hello_PortTypeClient client,
+            Func<Hello_PortType, Task<TResult>> action, ILogger logger = null) {
```

Comment:
In order to make this truly generic helper, we could do something like this

```
public static async Task<TResult> IssueSoapCallAsync<TClient, TResult>(ClientBase<TClient> client,
            Func<TClient, Task<TResult>> action, ILogger logger = null) where TClient : class
        {
```

Reply:
It is just more generic :)

Again, if you are OK with the suggested changes and also to make the call in HelloService more as an example (it can be just a commented duplicate method with the sample; or new `SayHelloWithHelper` method to demonstrate use of helper), let's continue to work on making these few changes and completing this PR.

Reply:
Sure, let's do it :)